src/projectile.py

```diff
@@ -144,20 +159,54 @@ def _shorten_beam(self, collision_list: list[LinkedSprite]):
         closest_collision: Optional[LinkedSprite] = None
         for collision in collision_list:
             collisions.append(collision)
-        point: Tuple[float, float] = self.muzzle_location[:2]
+        point: Tuple[float, float] = self.start_location[:2]
         point_vec: Tuple[float, float] = polar_to_cartesian(1, self.sprite.radians)
         for x in range(1, self.beam_range):
             if arcade.get_sprites_at_point(point, collisions):
                 closest_collision = arcade.get_sprites_at_point(point, collisions)[0]
                 self.sprite.width = x
+                self.hit_location = (point[0], point[1], self.sprite.radians)
                 break
             point = add_vec(point, point_vec)
         self._update_sprite_location()
         return closest_collision
 
     def _update_sprite_location(self):
-        set_sprite_location(self.sprite, self.muzzle_location)
-        move_sprite_polar(self.sprite, self.sprite.width / 2, self.muzzle_location[2])
+        set_sprite_location(self.sprite, self.start_location)
+        move_sprite_polar(self.sprite, self.sprite.width / 2, self.start_location[2])
+
+
+class Explosion(Projectile):
+    explosion_rate: float
+    explosion_radius: float
+    players_hit: list[LinkedSprite[Player]]
+
+    def setup(self, explosion_radius: float, explosion_rate: float):
+        self.explosion_radius = explosion_radius
+        self.explosion_rate = explosion_rate
+        self.sprite.height = 1
+        self.sprite.width = 1
+        self.players_hit = []
+
+    def update(self, delta_time: float):
+        if self.sprite.height < self.explosion_radius * 2:
+            self.sprite.height += self.explosion_rate * delta_time
+            self.sprite.width += self.explosion_rate * delta_time
+        else:
+            self.remove_sprite()
+
+    def on_collision_with_wall(
```

Comment:
Won't the explosion hurt players through walls?

If you want to use sprites for the explosion you could have the explosion start as one sprite and instead of growing have it propagate another sprite in each direction for X distance or if it hits a wall. Think Bomberman for reference

Reply:
Implemented. Explosions now consist of a bunch of SubExplosions going in every direction that stop at walls.

Reply:
Is something like "Shockwave" or "ShockwaveArc" a better name for `SubExplosion`?